File: solutions/javascript/triangle/1/triangle.js
//
// This is only a SKELETON file for the 'Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Triangle {
  constructor(...sides) {
    this.side1 = sides[0];
    this.side2 = sides[1];
    this.side3 = sides[2];
  }

  get isEquilateral() {
    return (this.side1 > 0 && this.side2 > 0 && this.side3 > 0) && this.side1 === this.side2 && this.side1 === this.side3;
  }

  get isIsosceles() {
    return this.side1 > 0 &&
      this.side2 > 0 &&
      this.side3 > 0 &&
      this.side1 < this.side2 + this.side3 &&
      this.side2 < this.side1 + this.side3 &&
      this.side3 < this.side1 + this.side2 &&
      (
        this.side1 === this.side2 ||
        this.side1 === this.side3 || 
        this.side2 === this.side3
      );
  }

  get isScalene() {
     return this.side1 > 0 &&
      this.side2 > 0 &&
      this.side3 > 0 &&
      this.side1 < this.side2 + this.side3 &&
      this.side2 < this.side1 + this.side3 &&
      this.side3 < this.side1 + this.side2 &&
      this.side1 != this.side2 && this.side1 != this.side3 && this.side2 != this.side3;
  }

}

File: solutions/java/triangle/1/src/main/java/Triangle.java
class Triangle {
    private double side1, side2, side3;
    
    Triangle(double side1, double side2, double side3) throws TriangleException {
        if(!isTriangle(side1, side2, side3)) {
            throw new TriangleException();
        }
        this.side1 = side1;
        this.side2 = side2;
        this.side3 = side3;
    }

    boolean isEquilateral() {
        return side1 == side2 && side1 == side3;
    }

    boolean isIsosceles() {
        return side1 == side2 || side1 == side3 || side2 == side3;
    }

    boolean isScalene() {
        return side1 != side2 && side1 != side3 && side2 !=  side3;
    }

    boolean isTriangle(double side1, double side2, double side3) {
        return side1 > 0 && side2 > 0 && side3 > 0 &&
            (side1 <= side2 + side3 && side2 <= side1 + side3 && side3 <= side1 + side2);
    }
}
